src/components/file-browser/FileBrowserDirectories.ts

```diff
@@ -0,0 +1,172 @@
+import { Fragment, ReactNode, useState } from 'react'
+import { details, div, h, li, p, span, summary, ul } from 'react-hyperscript-helpers'
+import { Link } from 'src/components/common'
+import { useDirectoriesInDirectory } from 'src/components/file-browser/file-browser-hooks'
+import { basename } from 'src/components/file-browser/file-browser-utils'
+import { icon } from 'src/components/icons'
+import FileBrowserProvider from 'src/libs/ajax/file-browser-providers/FileBrowserProvider'
+import colors from 'src/libs/colors'
+
+
+interface FileBrowserDirectoryContentsProps {
+  provider: FileBrowserProvider
+  isHighlighted: (path: string) => boolean
+  level: number
+  path: string
+  onSelectDirectory: (path: string) => void
+}
+
+const renderDirectoryStatus = (level: number, children: ReactNode) => p({ style: { padding: '0.5rem', margin: `0 0 0.25rem ${level * 1.25}rem` } }, [children])
+
+export const FileBrowserDirectoryContents = (props: FileBrowserDirectoryContentsProps) => {
+  const { provider, isHighlighted, level, path, onSelectDirectory } = props
+
+  const {
+    state: { status, directories },
+    hasNextPage,
+    loadNextPage
+  } = useDirectoriesInDirectory(provider, path)
+
+  if (status === 'Error') {
```

Comment:
I'm curious, do we get enums with typescript, and if so would that be something to leverage for these sorts of comparisons?

Reply:
TypeScript does have enums, but we've elected to prefer unions of literals over them. For example:
```ts
const Status = 'Loading' | 'Ready' | 'Error'

if (status === 'Loading') { ... }
```
vs
```ts
enum Status {
  Loading,
  Ready,
  Error,
}

if (status === Status.Loading) { ... }
```

See https://www.bam.tech/article/should-you-use-enums-or-union-types-in-typescript for more information on the pros and cons of each.

We do get type checking here. For example, if I change this line to `if (status === 'error')` then TypeScript complains: > This condition will always return 'false' since the types '"Loading" | "Ready" | "Error"' and '"error"' have no overlap.